app.rb

```diff
@@ -2,123 +2,161 @@
-      print 'Specialisation?'
+      puts 'Specialisation: '
       specialisation = gets.chomp
-      people.push(Teacher.new(specialisation, age, name))
+      people_data = Teacher.new(specialisation, age, name)
+      handle_data(people_data, 'teacher')
     end
     puts 'Person added successfully'
   end
 
-  def add_book(books)
+  def add_book
     puts 'Please Add a book'
     print 'Title :'
     title = gets.chomp.capitalize
     print 'Author:'
     author = gets.chomp.capitalize
-    books.push(Book.new(title, author))
+    book_info = Book.new(title, author)
+    book_data_hash = { title: book_info.title, author: book_info.author }
+    book_data = get_data('books')
+    book_data.push(book_data_hash)
+    update_data('books', book_data)
     puts 'Book added successfully'
   end
 
-  def add_rental(books, people, rental_data)
+  def add_rental
     puts 'Please select abook from the list by number'
-    books.map.with_index { |book, index| puts "#{index} Title: #{book.title}',Auther:#{book.author}" }
+    book.map.with_index { |item, index| puts "#{index} Title: #{item.title}',Auther:#{item.author}" }
     selected_book = gets.chomp.to_i
     puts "Choose a person from the list:(
       DON'T CHOOSE ID PLEASE)"
-    people.map.with_index do |person, index|
-      puts "#{index}, Name: #{person.name} Age: #{person.age},
-          ID:#{person.id}"
+    person.map.with_index do |item, index|
+      puts "#{index}, Name: #{item.name} Age: #{item.age},
+          ID:#{item.id}"
     end
     selected_person = gets.chomp.to_i
 
     print 'date?'
     selecteted_date = gets.chomp
-    rental_data.push(Rental.new(selecteted_date, people[selected_person], books[selected_book]))
     puts 'rental_data updated'
+    rental_data_hash = { date: selecteted_date, book_index: selected_book, person_index: selected_person }
+    rental_data = get_data('rental')
+    rental_data.push(rental_data_hash)
+    update_data('rental', rental_data)
   end
 
-  def show_rental_data(rentals)
+  def show_rental
     print 'Person ID'
     selected_person_id = gets.chomp.to_i
-    rentals.each do |rental|
-      next unless rental.person.id == selected_person_id
+    @rental.each do |item|
+      next unless item.person.id.to_i == selected_person_id
 
-      puts rental.book.title
-      puts rental.person.name
-      puts rental.date
+      puts item.book.title
+      puts item.person.name
+      puts item.date
     end
   end
 
   def display_all
     puts 'Welcome to the school library'
     puts ' Please choose a task  basing on the number '
     puts "1:  Show all books.
-            2:  Show all people.
-            3: Create a person
-            4: Create a rental
-            5: Create a book
-            6: List rented books to a person by ID
-            7: Exit"
+          2:  Show all people.
+          3: Create a person
+          4: Create a rental
+          5: Create a book
+          6: List rented books to a person by ID
+          7: Exit"
```

Comment:
- [ ] OPTIONAL
I'll urge you to kindly endeavor to improve on the following aspects of your UI for better user interaction/experience:
    * It will be a good idea to that your options (1- 6) and in a straight line.
    * It will be a good idea to swap option 4 with 5, to ensure a consistent logical flow of app.
    * Kindly ensure that all user inputs are displayed on the next line, to distinguish user inputs from queries.
    * Entered data are not displayed among the list of results, for instance when displaying all books or all persons. Please resolve this aspect.